_posts/2020-01-20-libmemkind.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Introduction to libmemkind
+author: jschmieg
+layout: post
+identifier: libmemkind
+---
+
+
+# Introduction
+
+One of the components of PMDK is libmemkind, a library which was developed to
+simplify
+usage of persistent memory in a volatile mode. There are NVDIMMs technologies,
+such as Intel Optane DCPMM, that provides persistency, byte-addressability, but 
+also a high capacity when compared with DRAM modules. By extension, they can be 
+used as an augmentation of main memory and utilized by applications which
+consume 
+a large amount of memory and do not require persistency, such as in-memory
+databases, 
+caching engines and scientific simulations.
+
+# Libmemkind
+
+According to the current persistent memory programming model, NVDIMMs are
+exposed by operating system as devices on which user should create file-system.
+This model creates a need for a way to consume memory exposed through files in 
+applications. Libmemkind fills this gap 
+by utilizing jemalloc on temporary files created on NVDIMMs and acts as a memory
+allocator for applications. The new option of utilizing persistent memory as a
+hotplugged memory region (KMEM DAX) is already available in Linux kernel
+beginning from
+version 5.1 and Memkind support for this feature will be described in the next
+blog entry.
+Libmemkind provides various memory pools called “kinds” for memories with
+miscellaneous characteristics: DRAM, persistent memory and High-Bandwidth
+Memory. This
+allows partitioning a heap of an application between these kinds. On a system
+equipped
+with DRAM and NVDIMMs, it is possible to re-design an application in a way 
```

Comment:
redesign is pretty heavy word for such change...

Reply:
done